llvm/lib/Transforms/IPO/MemProfContextDisambiguation.cpp

```diff
@@ -122,6 +122,20 @@ static cl::opt<unsigned>
                         cl::desc("Max depth to recursively search for missing "
                                  "frames through tail calls."));
 
+// By default enable cloning of callsites involved with recursive cycles
+static cl::opt<bool> AllowRecursiveCallsites(
+    "memprof-allow-recursive-callsites", cl::init(true), cl::Hidden,
+    cl::desc("Allow cloning of callsites involved in recursive cycles"));
+
+// When enabled, try to detect and prevent cloning of recursive contexts.
+// This is only necessary until we support cloning through recursive cycles.
+// Leave off by default for now, as it requires a little bit of compile time
+// overhead and doesn't affect correctness, it will just inflate the cold hinted
+// bytes reporting a bit when -memprof-report-hinted-sizes is enabled.
+static cl::opt<bool> SkipRecursiveContexts(
+    "memprof-skip-recursive-contexts", cl::init(false), cl::Hidden,
```

Comment:
Can you update this one to be positive intent too?

Reply:
done